node/src/accountant/mod.rs

```diff
@@ -381,43 +363,54 @@ impl Handler<NodeFromUiMessage> for Accountant {
                     client_id,
                     context_id,
                 },
-            );
+            )
         } else {
             handle_ui_crash_request(msg, &self.logger, self.crashable, CRASH_KEY)
         }
     }
 }
 
 impl Accountant {
-    pub fn new(
-        config: &BootstrapperConfig,
-        payable_dao_factory: Box<dyn PayableDaoFactory>,
-        receivable_dao_factory: Box<dyn ReceivableDaoFactory>,
-        pending_payable_dao_factory: Box<dyn PendingPayableDaoFactory>,
-        banned_dao_factory: Box<dyn BannedDaoFactory>,
-    ) -> Accountant {
+    pub fn new(config: &mut BootstrapperConfig, dao_factories: DaoFactories) -> Accountant {
+        let payment_thresholds = Rc::new(
+            config
+                .payment_thresholds_opt
+                .take()
+                .expectv("Payment thresholds"),
+        );
+        let scan_intervals = config.scan_intervals_opt.take().expectv("Scan Intervals");
+        let when_pending_too_long_sec = config
+            .when_pending_too_long_opt
+            .take()
+            .expectv("When pending too long sec");
+
+        let earning_wallet = Rc::new(config.earning_wallet.clone());
+        let financial_statistics = Rc::new(RefCell::new(FinancialStatistics::default()));
```

Comment:
I can empathise the nervousness that these Reference Counters create.

I pointed out the similar (if not the same) for this variable with Dan, and proposed an idea to split the field, that can possibly prevent us to not use the reference counters for this variable.

But Dan pointed out, that it's better to use the Reference Counters because of the shared responsibilities. There is a tradeoff and we want to pay more importance that the fields with shared responsibilities are owned by individual entities.